crates/core/src/auth/identity.rs

```diff
@@ -20,6 +22,105 @@ pub struct SpacetimeIdentityClaims {
     pub exp: Option<SystemTime>,
 }
 
+// The new token format that we are sending out.
+#[serde_with::serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SpacetimeIdentityClaims2 {
+    #[serde(rename = "hex_identity")]
+    pub identity: Identity,
```

Comment:
Do we still want to keep the `hex_identity` at all? It's derived from the subject and issuer, right? I suppose it doesn't hurt, but if we do keep it around it should probably be verified to be correctly computed by the subject and issuer.

I would advocate for just removing it.

Reply:
If we do keep it, I think it should be serialized as `id` or `identity` rather than `hex_identity`

Reply:
This does verify that the identity matches. I'm not sure if we need it to be in the token, but we should send the identity to clients to avoid needing to compute it on the client side.

Reply:
If we want to send it to clients, might as well put it in the token then.